core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -129,22 +127,29 @@ void start() {
       EnumSet<CacheContainerAdmin.AdminFlag> adminFlags = EnumSet.noneOf(CacheContainerAdmin.AdminFlag.class);
       persistedCaches.forEach((name, configuration) -> {
          ensurePersistenceCompatibility(name, configuration, persistedCaches);
+         // First create the cache locally to ensure that it's available on startup
+         createCacheLocally(name, null, configuration, adminFlags);
          // The cache configuration was permanent, it still needs to be
          CompletionStages.join(getOrCreateCache(name, configuration, adminFlags));
       });
 
       persistedTemplates.forEach((name, configuration) -> {
          ensurePersistenceCompatibility(name, configuration, persistedTemplates);
+         // First create the cache locally to ensure that it's available on startup
+         createTemplateLocally(name, configuration, adminFlags);
          // The template was permanent, it still needs to be
          CompletionStages.join(getOrCreateTemplate(name, configuration, adminFlags));
       });
+
+      // Register the GlobalConfigurationStateListener after initial caches created to prevent duplicate writes
+      getStateCache().addListener(stateCacheListener);
```

Comment:
I'm not sure if it is safe to register the listener at the end. 
if node X is starting (for example, it is in the loop above) and the user creates a new cache (on node Y, via CLI), how node X knows there is a new cache if the listener is not registered?